crates/nix_rs/src/flake/metadata.rs

```diff
@@ -0,0 +1,24 @@
+use crate::command::{NixCmd, NixCmdError};
+
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+use super::url::FlakeUrl;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct FlakeMetadata {
+    pub path: PathBuf,
+}
+
+/// Runs `nix flake metadata json` in Rust
+pub async fn from_nix(cmd: &NixCmd, flake_url: &FlakeUrl) -> Result<FlakeMetadata, NixCmdError> {
```

Comment:
Be consistent when it comes to writing API code.

In the rest of the project, is `from_nix` in fact a top-level function?